src/data/types.d.ts

```diff
@@ -8,65 +11,318 @@ export type ProductSlug =
   | 'vault'
   | 'waypoint'
 
+/**
+ * Global configuration for a product. Used to hold data that's used on
+ * multiple pages.
+ */
 export interface ProductData {
+  /**
+   * The name of the product, in Proper Noun case. Used wherever the name of
+   * the product is presented to a human.
+   */
   name: string
+
+  /**
+   * The name of the product, in lower case (and without spaces). Used wherever
+   * the name of the product is presented to a computer. Used in things like
+   * determining colors and other thematic elements; connecting to our content
+   * API; etc.
+   */
   slug: ProductSlug
+
+  /**
+   * The configuration for our Algolia-powered docs search component.
+   * @see {@link AlgoliaConfig}
+   */
   algoliaConfig: AlgoliaConfig
+
+  /**
+   * The configuration for analytics products such as Fathom and Segment.
+   * @see {@link AnalyticsConfig}
+   */
   analyticsConfig?: AnalyticsConfig
+
+  /**
+   * The configuration for <meta> tags.
+   * @see {@link Metadata}
+   */
   metadata: Metadata
+
+  /**
+   * Whether the AlertBanner should be displayed on every page.
+   */
   alertBannerActive: boolean
+
+  /**
+   * The configuration for the AlertBanner.
+   * @see {@link AlertBanner}
+   */
   alertBanner: AlertBanner
+
+  /**
+   * An array of items describing the primary top-level navigation.
+   * @see {@link SubnavItemClass}
+   */
   subnavItems: (SubnavItemClass | 'divider')[]
+
+  /**
+   * An allow-list of paths that this product uses for its documentation. Used
+   * to determine whether a relative URL should be resolved in the context of a
+   * product.
+   */
+  basePaths: string[]
+
+  /**
+   * The configuration for the sidebar rendered on pages for this product in
+   * Dev Portal.
+   * @see {@link Sidebar}
+   */
+  sidebar: Sidebar
+
+  /**
+   * An array of items describing the primary top-level navigation for this
+   * product in Dev Portal.
+   * @see {@link NavigationHeaderItem}
+   */
+  navigationHeaderItems: NavigationHeaderItem[]
```

Comment:
These are defined in another file and documenting them is part of another task I'm working on. They can be removed!